### PR TITLE
RC8 updates

### PR DESCRIPTION
**Pull request by EdNoepel**
**Changes**
- Updated ABIs to typechain for RC8 contracts.
- Updated to new RC8 testchain.
- Changes to bond factor and neutral price calculations.
- Removed _RewardsManager_ artifacts.

### CODE REVIEW

File: src/constants/common.ts [hunk 1]
```diff
@@ -2,8 +2,7 @@ import { BigNumber, utils } from 'ethers';
 import { DistributionPeriodStage } from '../types/classes';
 
 // transaction management
-export const GAS_MULTIPLIER = 2;
-export const GAS_LIMIT_MAX = 1500000;
```
Comment (EdNoepel) [inline src/constants/common.ts:6]: Higher multiplier was needed for some `repayDebt` transactions.  `GAS_LIMIT_MAX` was not used anywhere.

File: src/classes/Pool.ts [hunk 2]
```diff
@@ -503,21 +506,37 @@ export abstract class Pool {
...
+    return min(wmul(thresholdPrice, npTpRatio), MAX_INFLATED_PRICE_WAD);
+  }
+
+  /**
+   * calculate neutral price to threshold price "ratio"
+   * @param rate current pool "borrower" interest rate
+   * @returns relationship between the neutral price and threshold price
+   */
+  calculateNpTpRatio(rate: BigNumber) {
+    return toWad(1.04).add(wsqrt(rate).div(2));
```
Comment (FOr3st) [inline src/classes/Pool.ts:537]: What is this `1.04` constant? Do we want to extract and / or desribe?

**Review by imbaniac — APPROVED** [review]
Overall LGTM and approve, but I'd took into account @FOr3st comments.

File: src/classes/Pool.ts [hunk 3]
```diff
@@ -379,28 +379,29 @@ export abstract class Pool {
...
-    const [debt, collateral, t0np] = response[3];
-    const [kickTimestamp] = response[4];
+    const [, , , pendingInflator] = response[1];
+    const [debt, collateral, t0np] = response[2];
+    const [kickTimestamp] = response[3];
     const collateralization = debt.gt(0) ? collateral.mul(lup).div(debt) : toWad(1);
     const tp = collateral.gt(0) ? wdiv(debt, collateral) : BigNumber.from(0);
     const np = wmul(t0np, pendingInflator);
 
+    const rate = (await interestRateInfo(this.contract))[0];
```
[the remark this answers is not in the document]
Comment (EdNoepel) [reply src/classes/Pool.ts]: Improved in ba0aeffa82cfd54f1db1553e7908ad5eb4efccf3.

File: src/classes/Pool.ts [hunk 4]
```diff
@@ -428,9 +428,12 @@ export abstract class Pool {
     const retval = new Map<Address, Loan>();
     let i = 0;
     const [, , , , lup] = response[i];
-    const momp = BigNumber.from(response[++i]);
     const [, , , pendingInflator] = response[++i];
 
+    // calculate npTpRatio, needed for calculating liquidation bonds
+    const rate = (await interestRateInfo(this.contract))[0];
```
[the remark this answers is not in the document]
Comment (EdNoepel) [reply src/classes/Pool.ts]: Improved in ba0aeffa82cfd54f1db1553e7908ad5eb4efccf3.

File: src/classes/Pool.ts [hunk 5]
```diff
@@ -503,21 +506,37 @@ export abstract class Pool {
...
+  calculateLiquidationBond(npTpRatio: BigNumber, borrowerDebt: BigNumber) {
+    // bondFactor = min((NP-to-TP-ratio - 1)/10, 0.03)
+    const bondFactor = min(wdiv(npTpRatio.sub(toWad(1)), toWad(10)), toWad(0.03));
+    // bondSize_ = Maths.wmul(bondFactor_, borrowerDebt_);
     return wmul(bondFactor, borrowerDebt);
   }
 
+  /**
+   * calculates a loan's neutral price
+   * @param debt loan debt
```
[the remark this answers is not in the document]
Comment (EdNoepel) [reply src/classes/Pool.ts]: Corrected in ba0aeffa82cfd54f1db1553e7908ad5eb4efccf3.

**Review by FOr3st — APPROVED** [review]

File: src/classes/Pool.ts (hunk 2 above)
Comment (FOr3st) [reply src/classes/Pool.ts:536]: Thanks, I think PR is good to go :)

**Comment by EdNoepel** [comment]
Just published as version 0.2.0.

**Pull request merged**